haas/api.py

```diff
@@ -967,6 +967,21 @@ def show_switch(switch):
     }, sort_keys=True)
 
 
+@rest_call('GET', '/switch/<switch>/port/<path:port>', Schema({
+    'switch': basestring, 'port': basestring}))
+def show_port(switch, port):
+    """show port details on a switch.
+
+    Return a JSON object showing the node and nic to which port is connected.
+    """
+    get_auth_backend().require_admin()
+    switch = _must_find(model.Switch, switch)
+    port = _must_find_n(switch, model.Port, port)
+    nic = port.nic
+    return json.dumps({'node': nic.owner.label if nic else None, 'nic':
```

Comment:
I feel that the API shouldn't return a `None`. I am returning a an empty object `{}`.

Also, if I return a None, then whatever is making that call can't blindly do [response.loads()](https://github.com/CCI-MOC/hil/blob/master/haas/client/base.py#L44) because then it says that "No JSON object could be decoded".